rpc/internal/perf_service.go

```diff
@@ -228,21 +215,28 @@ func (srv *perfService) CloseMetrics(ctx context.Context, end *MetricsSeriesEnd)
 		return nil, newRPCError(codes.Internal, errors.Wrapf(err, "closing perf result record '%s'", record.ID))
 	}
 
-	resp.Success = true
-	return resp, nil
-}
+	hasEventData, err := srv.addFTDCRollupsJob(ctx, record.ID, record.Artifacts)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating FTDC rollups job")
+	}
 
-func (srv *perfService) addArtifacts(ctx context.Context, record *model.PerformanceResult, artifacts []*ArtifactInfo) error {
-	for _, a := range artifacts {
-		record.Artifacts = append(record.Artifacts, *a.Export())
+	// Only enqueue a new update time series job for mainline results if
+	// there are user-submitted rollups AND no event data (FTDC artifact).
+	// In the latter case, the FTDC rollups job will take care of
+	// enqueueing the update time series job.
+	if record.Info.Mainline && len(record.Rollups.Stats) > 0 && !hasEventData {
```

Comment:
Just to confirm, this is a deliberate decision to only include Mainline since that's all that the Signal Processing Service handles, right?

Reply:
Yes, the existing behavior already does this. I only added the last condition which is to check if the FTDC job was already enqueued.